### PR TITLE
Range parsers with tests b

### PR DESCRIPTION
**Pull request by Gerych1984**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

**Review by darkdef — APPROVED** [review]

**Review by terabytesoftw — APPROVED** [review]

File: src/RangeParser.php [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        }
+
+        if ($max !== null && $includeUpper === false) {
+            $max -= 1;
+        }
+
+        return [$min, $max];
+    }
+
+    private static function parseNumRange(?string $lower, ?string $upper, bool $includeLower, bool $includeUpper): array
```
Comment (terabytesoftw) [inline src/RangeParser.php:110]: `bool $includeLower, bool $includeUpper` they are not used within the method.

File: src/RangeParser.php [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        }
+
+        if ($max && $includeUpper === false) {
+            $max->sub($interval);
+        }
+
+        return [$min, $max];
+    }
+
+    private static function parseTsRange(?string $lower, ?string $upper, bool $includeLower, bool $includeUpper): array
```
Comment (terabytesoftw) [inline src/RangeParser.php:135]: `bool $includeLower, bool $includeUpper` they are not used within the method.

File: src/RangeParser.php [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    {
+        $min = $lower ? DateTime::createFromFormat('Y-m-d H:i:s', $lower) : null;
+        $max = $upper ? DateTime::createFromFormat('Y-m-d H:i:s', $upper) : null;
+
+        return [$min, $max];
+    }
+
+    private static function parseTsTzRange(?string $lower, ?string $upper, bool $includeLower, bool $includeUpper): array
```
Comment (terabytesoftw) [inline src/RangeParser.php:143]: `bool $includeLower, bool $includeUpper` they are not used within the method.

File: src/RangeParser.php (hunk 1 above)
Comment (Gerych1984) [reply src/RangeParser.php:110]: I will fix that and add new short immutable methods like `withType`, `asInt`, `asNum` etc

**Comment by samdark** [comment]
@Gerych1984 do you have time to rebase this PR and adjust it based on comments?

**Comment by Gerych1984** [comment]
> @Gerych1984 do you have time to rebase this PR and adjust it based on comments?

Yes. I haven't forgotten about this and [this](https://github.com/yiisoft/db-pgsql/pull/153) PR. I'll try to find time to get them in the right shape

**Pull request closed without merging**